core/src/main/java/org/mule/DefaultMuleMessage.java

```diff
@@ -224,6 +246,18 @@ public DefaultMuleMessage(Object message, MuleMessage previous, MuleContext mule
         resetAccessControl();
     }
 
+    private MuleMessageDataTypeWrapper<?> createDataTypeWrapper(DataType<?> dataType)
+    {
+        if (dataType instanceof MuleMessageDataTypeWrapper)
+        {
+            return new MuleMessageDataTypeWrapper<>(this, ((MuleMessageDataTypeWrapper) dataType).getDelegate());
```

Comment:
Does this happen somewhere, or it's just incase?

I assume you can't just return 'dataType' because it may have a diferent message instance?